src/modules/launcher/Wox.Infrastructure/Image/ImageLoader.cs

```diff
@@ -273,7 +274,7 @@ public static async ValueTask<ImageSource> LoadAsync(string path, bool generateT
                     if (GuidToKey.TryGetValue(hash, out string key))
                     {
                         // image already exists
-                        if (ImageCache.Usage.TryGetValue(path, out _))
+                        if (ImageCache.Usage.TryGetValue(path, out _) && Path.GetFileName(path).Equals(Path.GetFileName(key), StringComparison.Ordinal))
```

Comment:
Looking at the code, this seems to be using only the filename and extension part of the path. Why not the full path? Files with the same name might be present on different paths.